src/main/java/org/apache/mesos/offer/TaskUtils.java

```diff
@@ -94,6 +93,61 @@ public static TaskInfo clearTransient(TaskInfo taskInfo) {
     return taskBuilder.build();
   }
 
+  /**
+   * Sets a {@link Label} indicating the target configuruation for the provided {@link TaskInfo}.
+   * @param taskInfo is the TaskInfo which will have the appropriate configuration {@link Label} set.
+   * @param targetConfigurationId is the ID referencing a particular Configuration in the {@link ConfigStore}
+   * @return
+   */
+  public static TaskInfo setTargetConfiguration(TaskInfo taskInfo, UUID targetConfigurationId) {
+    taskInfo = clearConfigTarget(taskInfo);
+    Labels labels = Labels.newBuilder(taskInfo.getLabels())
+            .addLabels(
+                    Label.newBuilder()
+                    .setKey(TARGET_CONFIGURATION_KEY)
+                    .setValue(targetConfigurationId.toString())
+                    .build())
+            .build();
+
+    return TaskInfo.newBuilder(taskInfo)
+            .setLabels(labels)
+            .build();
+  }
+
+  /**
+   * Returns the ID referencing a configuration in a {@link ConfigStore} associated with the provided {@link TaskInfo}.
+   * @param taskInfo is the TaskInfo from which the the configuration ID will be extracted.
+   * @return the ID of the target configuration for the provided {@link TaskInfo}
+   * @throws TaskException when a TaskInfo is provided which does not contain a {@link Label} with an indicated target
+   * configuration
+   */
+  public static UUID getTargetConfiguration(TaskInfo taskInfo) throws TaskException {
+    for (Label label : taskInfo.getLabels().getLabelsList()) {
+      if (label.getKey().equals(TARGET_CONFIGURATION_KEY)) {
+        return UUID.fromString(label.getValue());
+      }
+    }
+
+    throw new TaskException("TaskInfo does not contain label with key: " + TARGET_CONFIGURATION_KEY);
+  }
+
+  private static TaskInfo clearConfigTarget(TaskInfo taskInfo) {
```

Comment:
Nit: the term is reversed here ("ConfigTarget" vs "TargetConfiguration" elsewhere) and could maybe reference more clearly the fact that it's removing labels? Maybe `clearTargetConfigLabel`.